compiler/rustc_middle/src/mir/interpret/error.rs

```diff
@@ -530,6 +530,7 @@ pub enum InterpError<'tcx> {
     /// Stop execution for a machine-controlled reason. This is never raised by
     /// the core engine itself.
     MachineStop(Box<dyn MachineStopType>),
+    Interrupted,
```

Comment:
I don't think this warrants a new top-level error category. Conceptually this feels like ResourceExhaustion to me -- like an execution step limit.

Reply:
🤔 

I was also previously wary of mentioning Ctrl-C too much because this actually just responds to a signal which I at least have manually generated in other ways.

(We also respond differently to the second Ctrl-C)

Reply:
Yea, I was mostly kidding. I think it's fine as a top level variant.